src/compiler/parser.ts

```diff
@@ -7281,7 +7281,7 @@ namespace ts {
             const pos = getNodePos();
             const name = tokenIsIdentifierOrKeyword(token()) ? parseIdentifierName() : parseLiteralLikeNode(SyntaxKind.StringLiteral) as StringLiteral;
             parseExpected(SyntaxKind.ColonToken);
-            const value = parseLiteralLikeNode(SyntaxKind.StringLiteral) as StringLiteral;
+            const value = parseAssignmentExpressionOrHigher();

```

Comment:
I changed this to `parseAssignmentExpressionOrHigher` since it's what objects parse out for values, so covers the most likely set of expressions people may mistakenly put here (which is to say: any expression except comma expressions). There are other common "mistakes" that people may make for import assertions - passing an arbitrary expression instead of an object-like-thing, using computed property names or shorthand or rest properties... But handling those gracefully would require more involved alterations to how we parse import assertions.